recipe/build.sh

```diff
@@ -1,6 +1,27 @@
 #!/bin/bash
 
-set -euo pipefail
+set -eox pipefail
 
-cd ${SRC_DIR}
-IPOPT_DIR=${PREFIX} ${PYTHON} -m pip install . -vv
+export PKG_CONFIG_PATH=$PKG_CONFIG_PATH:$PREFIX/lib/pkgconfig:$BUILD_PREFIX/lib/pkgconfig
+
+EXTRA_FLAGS=""
+NP_INC="${SP_DIR}/numpy/core/include/"
+echo "PYTHON TARGET=${PYTHON}"
+
+if [[ $CONDA_BUILD_CROSS_COMPILATION == "1" ]]; then
+  # Add pkg-config to cross-file binaries since meson will disable it
+  # See https://github.com/mesonbuild/meson/issues/7276
+  echo "[binaries]" >> "$BUILD_PREFIX"/meson_cross_file.txt
+  echo "pkg-config = '$(which pkg-config)'" >> "$BUILD_PREFIX"/meson_cross_file.txt
+  # Use Meson cross-file flag to enable cross compilation
+  EXTRA_FLAGS="--cross-file $BUILD_PREFIX/meson_cross_file.txt"
+  NP_INC=""
+fi
+
+PKG_CONFIG=$(which pkg-config)
+export PKG_CONFIG
+
+cd "${SRC_DIR}"
+export MESON_ARGS="-Dipopt_dir=${PREFIX} -Dincdir_numpy=${NP_INC} -Dpython_target=${PYTHON} ${EXTRA_FLAGS}"
+python -m build -n -x .
```

Comment:
- `MESON_ARGS` is an environment variable that is used within setup.py
- `python -m build` builds the wheel and is explained in my compilation docs that I added, I think.
- `pip install` then installs the wheel hence all the extra args

First it makes sense to explain the `python -m build` reasoning. The typical `pip install .` isn't very flexible and wasn't working with the new `meson` way of doing things and I think it has to do with where conda actually does the build.  `meson` was putting build artifacts somewhere pip couldn't find for whatever reason. I don't remember the specifics but `python -m build` ensured all the build files were kept where pip could find them with the subsequent `pip install ...` call. Also, `python -m build` is now the recommended approach for building Python packages [source](https://packaging.python.org/en/latest/tutorials/packaging-projects/)

The extra `MESON_ARGS` are used to tell `meson` where to search for build dependencies. Note that `meson` is dependent on python and numpy is needed on the build machine for converting fortran files. Knowing that and that it is not guaranteed in the `conda-forge` CI to be the same version or even architecture as the target machine, we need to specify the extra args so `meson` doesn't accidentally pick up the dependencies on the build machine and actually finds the correct target dependencies. `ipopt_dir` could be specified using the environment variable, but seemed unnecessary since we need to pass the other extra arguments anyway. It's mostly a quirk of conda-forge's cross-compilation infrastructure.

I hope i explained that well enough...

Reply:
I will add comments also when I fix the other things

Reply:
This all makes sense, thanks!